File: packages/sdkwork-utils-java/src/main/java/com/sdkwork/utils/OptionalUtils.java
package com.sdkwork.utils;

public final class OptionalUtils {
    private OptionalUtils() {
    }

    public static String coalesce(String... values) {
        for (String value : values) {
            if (!StringUtils.isBlank(value)) {
                return value.trim();
            }
        }
        return null;
    }

    public static String defaultIfBlank(String value, String defaultValue) {
        return StringUtils.isBlank(value) ? defaultValue : value.trim();
    }
}

File: packages/sdkwork-utils-kotlin/src/main/kotlin/com/sdkwork/utils/OptionalUtils.kt
package com.sdkwork.utils

object OptionalUtils {
    fun coalesce(vararg values: String?): String? {
        for (value in values) {
            if (!StringUtils.isBlank(value)) {
                return value!!.trim()
            }
        }
        return null
    }

    fun defaultIfBlank(value: String?, defaultValue: String): String {
        return if (StringUtils.isBlank(value)) defaultValue else value!!.trim()
    }
}
